tests/workspaces.rs

```diff
@@ -0,0 +1,91 @@
+use near_units::parse_near;
+use serde_json::json;
+use std::env;
+use std::process::Command;
+use workspaces::AccountId;
+
+#[tokio::test]
+async fn test_deploy_contract_self_upgrade() -> anyhow::Result<()> {
+    const CONTRACT_ACCOUNT: &str = "devgovgigs.near";
+    let worker = workspaces::sandbox().await?;
+
+    let mainnet = workspaces::mainnet_archival().await?;
+    let contract_id: AccountId = CONTRACT_ACCOUNT.parse()?;
+
+    //To use block height if needed
+    //const BLOCK_HEIGHT: BlockHeight = 97416242;
+
+    let contract = worker
+        .import_contract(&contract_id, &mainnet)
+        .initial_balance(parse_near!("1000 N"))
+        //.block_height(BLOCK_HEIGHT)
+        .transact()
+        .await?;
+
+    println!("{}", format!("{:?}", contract));
+
+    let outcome = contract
+        .call("new")
+        .args_json(json!({}))
+        .transact() // note: we use the contract's keys here to sign the transaction
+        .await?;
+
+    assert!(outcome.is_success());
+    assert!(format!("{:?}", outcome).contains("Migrated to version:"));
+
+    println!("Init outcome: {:#?}", outcome);
+
+    //compile the current code
```

Comment:
Before moving to the new version, add all types of posts and at least one community, and then after migration, check that get-methods work.

Reply:
Okay, will work on this.